### PR TITLE
Revert "CI: xfail tests failing on numpy dev"

### LINKED ISSUE #44343: CI: New Numpydev version broke ci
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
Two tests are affected
```

### Issue Description

```

    @pytest.mark.parametrize("quantile", [0.0, 0.1, 0.45, 0.5, 1])
    @pytest.mark.parametrize(
        "interpolation", ["linear", "lower", "higher", "nearest", "midpoint"]
    )
    @pytest.mark.parametrize(
        "data",
        [
            [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0],
            [8.0, 1.0, 3.0, 4.0, 5.0, 2.0, 6.0, 7.0],
            [0.0, np.nan, 0.2, np.nan, 0.4],
            [np.nan, np.nan, np.nan, np.nan],
            [np.nan, 0.1, np.nan, 0.3, 0.4, 0.5],
            [0.5],
            [np.nan, 0.7, 0.6],
        ],
    )
    def test_rolling_quantile_interpolation_options(quantile, interpolation, data):
        # Tests that rolling window's quantile behavior is analogous to
        # Series' quantile for each interpolation option
        s = Series(data)
    
        q1 = s.quantile(quantile, interpolation)
        q2 = s.expanding(min_periods=1).quantile(quantile, interpolation).iloc[-1]
    
        if np.isnan(q1):
            assert np.isnan(q2)
        else:
>           assert q1 == q2
E           assert 0.040000000000000036 == 0.04000000000000001

pandas/tests/window/test_rolling.py:1664: AssertionError
________ test_rolling_quantile_interpolation_options[data2-linear-0.45] ________
[gw0] linux -- Python 3.9.7 /usr/share/miniconda/envs/pandas-dev/bin/python

quantile = 0.45, interpolation = 'linear', data = [0.0, nan, 0.2, nan, 0.4]

    @pytest.mark.parametrize("quantile", [0.0, 0.1, 0.45, 0.5, 1])
    @pytest.mark.parametrize(
        "interpolation", ["linear", "lower", "higher", "nearest", "midpoint"]
    )
    @pytest.mark.parametrize(
        "data",
        [
            [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0],
            [8.0, 1.0, 3.0, 4.0, 5.0, 2.0, 6.0, 7.0],
            [0.0, np.nan, 0.2, np.nan, 0.4],
            [np.nan, np.nan, np.nan, np.nan],
            [np.nan, 0.1, np.nan, 0.3, 0.4, 0.5],
            [0.5],
            [np.nan, 0.7, 0.6],
        ],
    )
    def test_rolling_quantile_interpolation_options(quantile, interpolation, data):
        # Tests that rolling window's quantile behavior is analogous to
        # Series' quantile for each interpolation option
        s = Series(data)
    
        q1 = s.quantile(quantile, interpolation)
        q2 = s.expanding(min_periods=1).quantile(quantile, interpolation).iloc[-1]
    
        if np.isnan(q1):
            assert np.isnan(q2)
        else:
>           assert q1 == q2
E           assert 0.18000000000000005 == 0.18000000000000002
```

### Expected Behavior

Passing

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 0861da680b50accb312d9dc8f422aa65bcdaa2b8
python           : 3.8.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.11.0-38-generic
Version          : #42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.0.dev0+1056.g0861da680b
numpy            : 1.22.0.dev0+1693.gf52acb53a
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 58.0.4
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.23.1
sphinx           : 4.2.0
blosc            : None
feather          : None
xlsxwriter       : 3.0.1
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.28.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
fsspec           : 2021.11.0
fastparquet      : 0.7.1
gcsfs            : 2021.05.0
matplotlib       : 3.4.3
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : 2021.11.0
scipy            : 1.7.1
sqlalchemy       : 1.4.25
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.18.2
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.53.1
None

Process finished with exit code 0

</details>

### PR DESCRIPTION
**Pull request by lithomas1**
Reverts pandas-dev/pandas#44362
closes #44343

### CODE REVIEW

File: pandas/tests/arrays/sparse/test_array.py [hunk 1]
```diff
@@ -1255,7 +1255,7 @@ def test_to_coo(
         A, rows, cols = ss.sparse.to_coo(
             row_levels=(0, 1), column_levels=(2, 3), sort_labels=sort_labels
         )
-        assert isinstance(A, scipy.sparse.coo_matrix)
+        assert isinstance(A, scipy.sparse.coo.coo_matrix)
```
Comment (phofl) [inline pandas/tests/arrays/sparse/test_array.py:1258]: This should be kept I think? Same one above?

Comment (phofl) [reply pandas/tests/arrays/sparse/test_array.py:1258]: Or did they change that back too?

Comment (alimcmaster1) [reply pandas/tests/arrays/sparse/test_array.py:1258]: Yeah agree this should be kept as is

**Review by alimcmaster1 — APPROVED** [review]
LGTM - we should revert the backport too right :)

**Pull request merged**